nevergrad/optimization/optimizerlib.py

```diff
@@ -1492,24 +1492,32 @@ def __init__(
             budget: Optional[int] = None,
             num_workers: int = 1,
             isotropic: bool = True,
-            naive: bool = True
+            naive: bool = True,
+            population_size_adaptation: bool = False
     ) -> None:
         super().__init__(parametrization, budget=budget, num_workers=num_workers)
         self.isotropic: bool = isotropic
         self.naive: bool = naive
+        self.population_size_adaptation = population_size_adaptation
+        self.min_coef_parallel_context: int = 8
+        # Sigma initialization
         self.sigma: tp.Union[float, np.ndarray]
         if self.isotropic:
             self.sigma = 1.0
         else:
             self.sigma = np.ones(self.dimension)
-        self.mu = max(16, self.dimension)
-        self.llambda = 4 * self.mu
-        if budget is not None and self.llambda > budget:
-            self.llambda = budget
-            self.mu = self.llambda // 4
-            warnings.warn("Budget may be too small in front of the dimension for EMNA")
-        if num_workers is not None:
-            self.llambda = max(self.llambda, num_workers)
+        # population size and parent size initializations
+        dim = self.dimension
+        self.popsize = _PopulationSizeController(llambda=4 * dim, mu=dim, dimension=dim, num_workers=num_workers)
+        if not self.population_size_adaptation:
+            self.popsize.mu = max(16, self.dimension)
+            self.popsize.llambda = 4 * self.popsize.mu
+            if budget is not None and self.popsize.llambda > budget:
+                self.popsize.llambda = budget
+                self.popsize.mu = self.popsize.llambda // 4
+                warnings.warn("Budget may be too small in front of the dimension for EMNA")
```

Comment:
can you send a base.InefficientSettingsWarning instead?

https://github.com/facebookresearch/nevergrad/blob/b01c503748ed2dbbe0c3e0bc4f49baf2964c458a/nevergrad/optimization/differentialevolution.py#L87-L88

This makes the warning uniform within nevergrad, and I filter them out during unit tests